File: backend/src/main/java/com/b4s/backend/services/StudentService.java
package com.b4s.backend.services;

import com.b4s.backend.api.dto.StudentDTO;
import com.b4s.backend.domain.Student;

import java.util.List;


public interface StudentService {
    Student getStudentByCpf(String cpf);

    void create(Student student);

    void delete(String cpf);

    List<Student> getAllStudents();

    Long getTotalStudents();

    void update(StudentDTO dto, String cpf);

    void delegateNewBus(String studentCpf, String busPlate);

    void removeBusFromStudent(String cpf, String plate);
}

File: frontend/src/services/StudentService.ts
import { Api } from '../providers'
import { Student, StudentDTO } from '../schemas'

export const config = { headers: { 'Content-Type': 'text' } }

const getAllStudents = () => Api.get<Student[]>('/student')

const getStudentByCpf = (cpf: string) => Api.get<Student>(`/student/${cpf}`)

const deleteStudent = (cpf: string) => Api.delete(`/student/${cpf}`)

const getTotalStudents = () => Api.get<number>('/student/total')

const insertNewStudent = (student: StudentDTO) => Api.post('/student', student)

const delegateBus = (cpf: string, plate: string) =>
  Api.post(`/student/delegate/${cpf}/${plate}`)

const removeBusFromStudent = (cpf: string, plate: string) =>
  Api.delete(`/student/delegate/${cpf}/${plate}`)

const updateStudent = (cpf: string, student: StudentDTO) =>
  Api.put(`/student/${cpf}`, student)

export const StudentService = {
  getAllStudents,
  getStudentByCpf,
  deleteStudent,
  getTotalStudents,
  insertNewStudent,
  delegateBus,
  removeBusFromStudent,
  updateStudent
}
